src/js/live-tracker.js

```diff
@@ -0,0 +1,201 @@
+import Component from './component.js';
+
+/* track when we are at the live edge, and other helpers for live playback */
+class LiveTracker extends Component {
+
+  constructor(player, options, ready) {
+    super(player, options, ready);
+
+    this.on(this.player_, 'durationchange', this.handleDurationchange);
+  }
+
+  isBehind() {
+    const liveCurrentTime = this.liveCurrentTime();
+    const currentTime = this.player_.currentTime();
+    const segmentLength = this.segmentLength_ || 12;
+
+    // the live edge window is the amount of seconds away from live
+    // that a player can be, but still be considered live.
+    // we add 0.07 because the live tracking happens every 30ms
+    // and we want some wiggle room for short segment live playback
+    const liveEdgeWindow = segmentLength + 0.07;
+
+    return liveCurrentTime !== Infinity && (liveCurrentTime - liveEdgeWindow) >= currentTime;
+  }
+
+  // all the functionality for tracking when seek end changes
+  // and for tracking how far past seek end we should be
+  trackLive_() {
+    this.pastSeekEnd_ = this.pastSeekEnd_ || 0;
+    const seekable = this.player_.seekable();
+
+    // skip undefined seekable
+    if (!seekable || !seekable.length) {
+      return;
+    }
+
+    const newSeekEnd = seekable.end(0);
+
+    // we can only tell if we are behing live, when seekable changes
+    // once we detect that seekable has changed we check the new seek
+    // end against current time, with a fudge value of half a second.
+    if (newSeekEnd !== this.lastSeekEnd_) {
+      if (this.lastSeekEnd_) {
+        this.segmentLength_ = newSeekEnd - this.lastSeekEnd_;
+      }
+
+      this.pastSeekEnd_ = 0;
+      this.lastSeekEnd_ = newSeekEnd;
+      this.trigger('seek-end-change');
+    }
+
+    this.pastSeekEnd_ = this.pastSeekEnd() + 0.03;
+
+    if (this.isBehind() !== this.behindLiveEdge()) {
+      this.behindLiveEdge_ = this.isBehind();
+      this.trigger('live-edge-change');
```

Comment:
This event naming seems contradictory to what we have in most places (e.g. no punctuation, all lowercase).

Reply:
yeah, I didn't think about that there is also `seek-end-change`. If we want to change it to `liveedgechange` and `seekendchange` we can do that.